### PR TITLE
Support allocating CPUs from shared pool and from exclusive pool

### LINKED ISSUE #17: Support containers asking for resources from multiple Pooler managed pools
**Is your feature request related to a problem? Please describe.**
Currently a container can only ask CPU pool managed resources from exactly one pool.
Theoretically it could be possible join a container to multiple pools.
This would benefit Pods which are used to supervise their own CPUs, and/or put multiple, different kind of processes into the same container for whatever reason. 

**Describe the solution you'd like**
An easy and straightforward solution would be simply joining the CPU sets of all requested pools into the final one.
Obviously this only works if the application knows about the drawbacks, and is actively managing the related risks itself.
For backward compatibility reasons a new config flag could be introduced to control this feature.

### PR DESCRIPTION
**Pull request by TimoLindqvist**
- Container can ask CPUs from one shared pool and from one exclusive
  pool. Allocating CPUs from two pools of same type is not supported

- Fixed example deployment manifests. Examples are using kube-system
  namespace for all CPU pooler components

Closes #17 , #19 

Signed-off-by: Timo Lindqvist <timo.lindqvist@nokia.com>

### CODE REVIEW

**Comment by Levovar** [comment]
LGTM on the Device Plugin side.
we could consider still adding limits to the mixed container with summa(exclusive + shared). not the best limit, but at least something

on the cpusetter side how do you imagine the solution: simple union of full shared set + the allocated exclusives?
random thought reg the environment variables: we push both shared and exclusive environment variables into the container in this case?

File: deployment/cpu-dev-ds.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: apps/v1
 kind: DaemonSet
```
Comment (maxcaresywwforever) [inline deployment/cpu-dev-ds.yaml]: cpu device daemon set also requires some proper role to be able to access Kubernetes "node" resource. Can this kind of change included here as well.
That change could fix this issue: https://github.com/nokia/CPU-Pooler/issues/19

**Comment by maxcaresywwforever** [comment]
Cluster role `caas:infra-psp` is NCIR specific. I think we need remove this from `deployment/cpusetter-ds.yaml`. Could we add this change to this PR as well since anyway the deployment yamls are changed here.

File: deployment/cpu-dev-ds.yaml (hunk 1 above)
Comment (TimoLindqvist) [reply deployment/cpu-dev-ds.yaml]: I'd rather keep changes to different issues in separate pull requests but to make the example deployment working, I'm adding the rbac configuration here.

**Comment by TimoLindqvist** [comment]
@Levovar It would be nice to set the limit but I'm afraid enabling the CPU bandwidth management will cause some small kernel processing in the CPUs where the container is running. If container has exclusive cpus, probably it is something latency critical so we should avoid enabling anything that could cause interrupts to the running process. Also I'm not sure how the CPU time is limited in case the thread that is running in shared CPU(s) exceeds the limit. Probably it will cause problems to the threads running in the exclusive CPUs.

Regarding the the `cpusetter` component. Plan is that the cpuset is a union of full shared set + the allocated exclusive cpus. And both shared and exclusive environment variables are set to the container. This actually happens already now without any changes to the cpu device plugin.

Added a commit with changes to `cpusetter`.

**Comment by Levovar** [comment]
@TimoLindqvist : yeah I understand. Probably the safest at this point to start with no limit, agree. CPUSetter changes look good!

@maxcaresywwforever: sure, will do! I also need to correct some mounts in the manifest anyway

**Comment by Levovar** [comment]
@maxcaresywwforever : didn't want to mess with Timo's review, so I updated the CPUSetter deployment manifests in a separate PR
https://github.com/nokia/CPU-Pooler/pull/23

**Comment by maxcaresywwforever** [comment]
@Levovar good to know. thanks.

**Comment by maxcaresywwforever** [comment]
No push, just want to know your plan for this PR to be merged?

**Comment by TimoLindqvist** [comment]
> 
> 
> No push, just want to know your plan for this PR to be merged?

Very soon. I'm hoping @Levovar could take a look also the latest changes.

**Comment by Levovar** [comment]
yeah I'm okay with it :) we can discuss making the enforced limits + request values uniform in the other issue I created for them, it is not burning

**Pull request merged**